packages/rocketchat-ui/client/lib/fileUpload.js

```diff
@@ -45,8 +45,104 @@ function formatBytes(bytes, decimals) {
 	return `${ parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) } ${ sizes[i] }`;
 }
 
-fileUpload = function(filesToUpload) {
+performFileUpload = function(file, fileName, fileDescription) {
```

Comment:
Why the modifications here? Couldn't we use the default file upload?

Reply:
@gdelavald, We are using the default file upload here still. What I did was generalize part of original fileUpload function so that it could be more reusable. You can see that the fileUpload function have calling "performFileUpload" function too so perform the file upload core activity. :-)